### PR TITLE
XRT-2109 Various updates

### PR DESCRIPTION
**Pull request by SteveOss**
Support added for string comparison function plus synchronous schedule execution
Fix added for compiling on MacOS

### CODE REVIEW

**Comment by SteveOss** [comment]
Note not required for XRT 2.1 release.

**Review by BinduRao2018 — COMMENTED** [review]
comments in-line

File: include/iot/scheduler.h [hunk 1]
```diff
@@ -120,13 +120,27 @@ extern iot_schedule_t * iot_schedule_create
...
+ * cannot be made concurrent.
  *
  * @param schedule Pointer to the schedule
  * @param enable   Whether to enable concurrent schedule execution
  */
 extern void iot_schedule_set_concurrent (iot_schedule_t * schedule, bool enable);
 
+/**
+ * @brief Enable synchronous execution of a schedule, where the main scheduling thread
+ * also executes the schedule function. Note that for repewated schedules, the repeat
```
[the remark this answers is not in the document]
Comment (SteveOss) [reply include/iot/scheduler.h]: Fix comitted

File: include/iot/data.h [hunk 2]
```diff
@@ -144,6 +144,9 @@ typedef void (*iot_data_free_fn) (void * ptr);
 /** Type for data update function pointer */
 typedef iot_data_t * (*iot_data_update_fn) (const iot_data_t * data, void * arg);
 
+/** Funcion to compare string data with a string value */
```
[the remark this answers is not in the document]
Comment (SteveOss) [reply include/iot/data.h]: Fix committed

File: include/iot/scheduler.h [hunk 3]
```diff
@@ -120,13 +120,27 @@ extern iot_schedule_t * iot_schedule_create
...
 
 /**
- * @brief Enable concurrent execution of a schedule
+ * @brief Enable concurrent execution of a schedule. Synchronous schedules
+ * cannot be made concurrent.
  *
  * @param schedule Pointer to the schedule
  * @param enable   Whether to enable concurrent schedule execution
  */
 extern void iot_schedule_set_concurrent (iot_schedule_t * schedule, bool enable);
```
[the remark this answers is not in the document]
Comment (SteveOss) [reply include/iot/scheduler.h:129]: Yes, but can't change signature for backwards compatibility. Really both concurrent/sync options should be set as "flags" on schedule creation, but have been added as separate functions to enable backwards compatibility. Should be re-worked for 1.5 release.

[the remark this answers is not in the document]
Comment (BinduRao2018) [reply include/iot/scheduler.h:129]: @SteveOss, what about this one?

**Review by BinduRao2018 — APPROVED** [review]
LGTM

**Pull request merged**